lib/src/frontend/future_matchers.dart

```diff
@@ -80,3 +80,54 @@ class _Completes extends AsyncMatcher {
     return description;
   }
 }
+
+/// Matches a [Future] that does not complete.
+///
+/// Note that this creates an asynchronous expectation. The call to
+/// `expect()` that includes this will return immediately and execution will
+/// continue. Later, after 20  passes through the event queue,
+/// the matcher will complete.
+///
+/// This returns an [AsyncMatcher], so [expect] won't complete until
+/// [timesToPump] passes through the event queue.
+final Matcher doesNotComplete = const _DoesNotComplete(20);
+
+/// Matches a [Future] that does not complete.
+///
+/// Note that this creates an asynchronous expectation. The call to
+/// `expect()` that includes this will return immediately and execution will
+/// continue. Later, after [timesToPump]  passes through the event queue,
+/// the matcher will complete.
+///
+/// This returns an [AsyncMatcher], so [expect] won't complete until
+/// [timesToPump] passes through the event queue.
+Matcher doesNotCompleteAfter(int timesToPump) =>
+    new _DoesNotComplete(timesToPump);
+
+class _DoesNotComplete extends AsyncMatcher {
+  final int timesToPump;
```

Comment:
Style thing: I'd personally make this private. It doesn't *really* matter, since the class itself is private, but whether a member is private is still a good signal for people reading the code as to whether it's accessed outside of the declaring class.